projects/smoke_bts7200/src/main.c

```diff
@@ -0,0 +1,80 @@
+// A simple smoke test for BTS7200
+
+// Periodically take reading from user selected channels and log the result
+// Configurable items: wait time, FRONT or REAR power distro selection, channels to be tested
+#include "bts_7200_current_sense.h"
+#include "current_measurement.h"
+#include "current_measurement_config.h"
+
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "mux.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// Smoke test settings. Can be modified to fit testing purpose.
+#define CURRENT_MEASURE_INTERVAL_MS 500  // Set wait time between each set of readings
+#define IS_FRONT_POWER_DISTRO true       // Set whether to test FRONT or REAR power distro
+
+// Number of channels to be tested, MUST match the number of elements in test_channels
+#define NUM_TEST_CHANNELS 8
+// Set of channels to be tested, the number of elements MUST match NUM_TEST_CHANNELS.
+// Range of channels: 0-7
+static uint8_t test_channels[NUM_TEST_CHANNELS] = { 0, 1, 2, 3, 4, 5, 6, 7 };
```

Comment:
As far as I can tell, the purpose of the `test_channels` array is to allow the smoketest user to exclude some BTS7200s from the test. In that case, you should make it clear that the "channels" are indices of BTS7200s in the front/rear HW config, and you should make the purpose of the array clear.

(Also, `test_channels` should be prefixed with `s_` since it's static, and I don't see a need for the `NUM_TEST_CHANNELS` define: can't you just use `SIZEOF_ARRAY(test_channels)` each time?)

Reply:
True I definitely should make it clearer what `test_channels` does. I always thought the size of array is required when declaring it, but I guess I could do `static uint_8 test_channels_s[] = {0,1,2,3,4,5,6,7}`